algorithm/index.mjs

```diff
@@ -1,17 +1,487 @@
+      cuota = intereses;
+      amortizacion = 0;
+      saldoFinal = saldoInicial + amortizacion;
+    } else {
+      cuota = -(
+        (Math.pow(1 + periodicalInterestRate, periodos - periodo + 1) * saldoInicial * periodicalInterestRate) /
+        (Math.pow(1 + periodicalInterestRate, periodos - periodo + 1) - 1)
+      );
+      amortizacion = cuota - intereses;
+      saldoFinal = saldoInicial + amortizacion;
+    }
+
+    console.log(`\nSaldo inicial: S/ ${roundMoney(saldoInicial)}`);
+    console.log(`Intereses: S/ ${roundMoney(intereses)}`);
+    console.log(`Cuota: S/ ${roundMoney(cuota)}`);
+    console.log(`Amortización: S/ ${roundMoney(amortizacion)}`);
+    console.log(`Monto de seguro contra todo riesgo: S/ ${roundMoney(insuranceAmount)}`);
+    console.log(`Costos periódicos: S/. ${roundMoney(periodicCosts)}`);
+
+    ahorroTributario = (intereses + insuranceAmount + periodicCosts + depreciacion) * IR;
+    montoIGVPeriodico = (cuota + insuranceAmount + periodicCosts) * IGV;
+    flujoBruto = cuota + insuranceAmount + periodicCosts;
+
+    if (inputs.buyingOption && periodo === periodos) {
+      montoIGVPeriodico = (cuota + insuranceAmount + periodicCosts + montoRecompra) * IGV;
+      flujoBruto = cuota + insuranceAmount + periodicCosts + montoRecompra;
+      console.log(`Monto de recompra: S/ ${roundMoney(montoRecompra)}`);
+    }
+
+    flujoIGV = flujoBruto + montoIGVPeriodico;
+    flujoNeto = flujoBruto - ahorroTributario;
+
+    vnaFlujoBruto += flujoBruto / Math.pow(1 + descuentoKs, periodo);
+    vnaFlujoNeto += flujoNeto / Math.pow(1 + descuentoWACC, periodo);
+
+    console.log(`Saldo final: S/ ${roundMoney(saldoFinal)}`);
+    console.log(`Depreciación: S/ ${roundMoney(depreciacion)}`);
+    console.log(`Ahorro tributario: S/ ${roundMoney(ahorroTributario)}`);
+    console.log(`IGV: S/ ${roundMoney(montoIGVPeriodico)}`);
+    console.log(`Flujo Bruto: S/ ${roundMoney(flujoBruto)}`);
+    console.log(`Flujo con IGV: S/ ${roundMoney(flujoIGV)}`);
+    console.log(`Flujo Neto:S/ ${roundMoney(flujoNeto)}\n`);
+
+    totalAmortizacion = totalAmortizacion + amortizacion;
+
+    if (!(tipoPlazo === "Total")) {
+      totalIntereses = totalIntereses + intereses;
+    }
+
+    totalCostosPeriodicos += periodicCosts;
+    totalSeguro += insuranceAmount;
+    saldoInicial = saldoFinal;
+
+    flujosBrutos.push(roundMoney(flujoBruto));
+    flujosNetos.push(roundMoney(flujoNeto));
+  }
+
+  const desembolsoTotal = totalIntereses + totalAmortizacion + totalSeguro + totalCostosPeriodicos + montoRecompra;
+
+  console.log(`Monto total por intereses: S/ ${roundMoney(-totalIntereses)}`);
+  console.log(`Monto total amortizado: S/ ${roundMoney(-totalAmortizacion)}`);
+  console.log(`Monto total por costos periódicos: S/ ${roundMoney(-totalCostosPeriodicos)}`);
+  console.log(`Monto total por seguro contra todo riesgo: S/ ${roundMoney(-totalSeguro)}`);
+  console.log(`Desembolso total: S/ ${roundMoney(-desembolsoTotal)}\n`);
```

Comment:
You can use the `Intl.NumberFormat` API here as well.

Reply:
Done